test/python/providers/test_pulse_defaults.py

```diff
@@ -0,0 +1,195 @@
+        self.assertEqual(self.ops_def.get_parameters('pv_test', 0), ('P1', 'P2'))
+
+        sched = self.ops_def.get('pv_test', 0, P1=0, P2=-1)
+        self.assertEqual(sched.instructions[0][-1].command.value, -1)
+        with self.assertRaises(PulseError):
+            self.ops_def.get('pv_test', 0, 0, P1=-1)
+        with self.assertRaises(PulseError):
+            self.ops_def.get('pv_test', 0, P1=1, P2=2, P3=3)
+
+    def test_sequenced_parameterized_schedule(self):
+        """Test parametrized schedule consists of multiple instruction. """
+        converter = QobjToInstructionConverter([], buffer=0)
+        qobjs = [PulseQobjInstruction(name='fc', ch='d0', t0=10, phase='P1'),
+                 PulseQobjInstruction(name='fc', ch='d0', t0=20, phase='P2'),
+                 PulseQobjInstruction(name='fc', ch='d0', t0=30, phase='P3')]
+        converted_instruction = [converter(qobj) for qobj in qobjs]
+
+        self.ops_def.add('inst_seq', 0, ParameterizedSchedule(*converted_instruction,
+                                                              name='inst_seq'))
+
+        with self.assertRaises(PulseError):
+            self.ops_def.get('inst_seq', 0, P1=1, P2=2, P3=3, P4=4, P5=5)
+
+        with self.assertRaises(PulseError):
+            self.ops_def.get('inst_seq', 0, P1=1)
+
+        with self.assertRaises(PulseError):
+            self.ops_def.get('inst_seq', 0, 1, 2, 3, P1=1)
+
+        sched = self.ops_def.get('inst_seq', 0, 1, 2, 3)
+        self.assertEqual(sched.instructions[0][-1].command.phase, 1)
+        self.assertEqual(sched.instructions[1][-1].command.phase, 2)
+        self.assertEqual(sched.instructions[2][-1].command.phase, 3)
+
+        sched = self.ops_def.get('inst_seq', 0, P1=1, P2=2, P3=3)
+        self.assertEqual(sched.instructions[0][-1].command.phase, 1)
+        self.assertEqual(sched.instructions[1][-1].command.phase, 2)
+        self.assertEqual(sched.instructions[2][-1].command.phase, 3)
+
+        sched = self.ops_def.get('inst_seq', 0, 1, 2, P3=3)
+        self.assertEqual(sched.instructions[0][-1].command.phase, 1)
+        self.assertEqual(sched.instructions[1][-1].command.phase, 2)
+        self.assertEqual(sched.instructions[2][-1].command.phase, 3)
+
+    def test_default_building(self):
+        """Test building of ops definition is properly built from backend."""
+        self.assertTrue(self.ops_def.has('u1', (0,)))
+        self.assertTrue(self.ops_def.has('u3', (0,)))
+        self.assertTrue(self.ops_def.has('u3', 1))
+        self.assertTrue(self.ops_def.has('cx', (0, 1)))
+        self.assertEqual(self.ops_def.get_parameters('u1', 0), ('P1',))
+        u1_minus_pi = self.ops_def.get('u1', 0, P1=1)
+        fc_cmd = u1_minus_pi.instructions[0][-1].command
+        self.assertEqual(fc_cmd.phase, -np.pi)
+        for chan in u1_minus_pi.channels:
+            # buffer no longer supported
+            self.assertEqual(chan.buffer, 0)
+
+    # def test_str(self):
```

Comment:
As mentioned above, I need to inspect the str outputs again, then I'll update this test.